QuizApp/App/Screens/Quiz session/ProgressView.swift

```diff
@@ -0,0 +1,85 @@
+import UIKit
+import Combine
+
+final class ProgressView: UIView {
+
+    private var progressViews: [UIView]!
+    private var stackView: UIStackView!
+    private var numberOfQuestions: Int
+
+    var questionNumber: Int = 1 {
+        didSet {
+            updateQuestion(for: questionNumber)
+        }
+    }
+
+    required init(numberOfQuestions: Int) {
+        self.numberOfQuestions = numberOfQuestions
+        super.init(frame: .zero)
+        createViews()
+        styleViews()
+        defineLayoutForViews()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+}
+
+extension ProgressView: ConstructViewsProtocol {
+
+    func createViews() {
+        progressViews = addViews()
+
+        stackView = UIStackView()
+        addSubview(stackView)
+    }
+
+    func styleViews() {
+        progressViews.forEach {
+            stackView.addArrangedSubview($0)
+            $0.backgroundColor = .white.withAlphaComponent(0.3)
+            $0.layer.cornerRadius = 3
+        }
+
+        stackView.axis = .horizontal
+        stackView.alignment = .fill
+        stackView.distribution = .fillEqually
+        stackView.spacing = 8
+    }
+
+    func defineLayoutForViews() {
+        stackView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+    }
+
+}
+
+private extension ProgressView {
+
+    func updateQuestion(for number: Int) {
+        var current = 0
+
+        while current < number {
+            progressViews[current].backgroundColor = .white
+            current += 1
+        }
```

Comment:
One way to avoid counter: 
```suggestion
        for (index, item) in progressViews.enumerated() {
            if index == number { break }
            progressViews[index].backgroundColor = .white
        }
```

But maybe you could also just color the view of the last answered question. So: 

```suggestion
     progressViews[number-1].backgroundColor = .white
```